internal/app/apiserver.go

```diff
@@ -63,22 +62,17 @@ func Run() {
 	}
 	defer pgClient.Close()
 	
-	repos := &repository.Repositories{
-		Courses: fake_repo.NewCourses(),
-		Users:   repository.NewUsersRepo(pgClient),
-	}
+	services := service.NewServices(service.Deps{
+		Repos: repository.NewRepositories(pgClient),
+		IdGen: idGen,
+	})
 	
 	bearerAuth, err := createAuthenticator(cfg)
 	if err != nil {
 		log.Fatal(err)
 	}
 	
```

Comment:
And this is the reverse side of the medal: things become more complicated here. As a next step, we could investigate [Wire](https://github.com/google/wire)

Reply:
Two things: 

1. Don't worry about this becoming more complex, is ok that this happens because we have to connect everything in this place, and as you said, it is the price to pay in order to have cleaner code.
2. I don't see the need to add a dependency to take care of dependencies. It is my opinion that is always better to keep the dependencies number as minimum as possible. If we were talking about a big project in which doing this configuration is a complex task then I'd think on using a library to bring this complexity down, but I don't think this project is the case.

Reply:
Now that I thought about it a little more, it is also ok if you want to use that library to experiment in this project. I was trying to give my point of view for a real life/workplace kind of scenario